compiler/AST/AggregateType.cpp

```diff
@@ -2361,7 +2508,8 @@ void AggregateType::fieldToArgType(DefExpr* fieldDef, ArgSymbol* arg) {
 
 bool AggregateType::addSuperArgs(FnSymbol*                    fn,
```

Comment:
This function really needs to be renamed. Could it be `handleSuperFields` ?

Reply:
Maybe ``addSuperFormals``?

Reply:
The issue is that it creates and adds a `super.init` call in the body of the function in addition to adding formal arguments. I think that's very unclear from the name `addSuperArgs`, which implies it is only adding arguments (even though it is unclear if they are actuals or formals).

I think `handleSuperFields` would address that, as would `addSuperFormalsAndSuperCall`.

Or we could decide it is not a problem for this PR to solve.